### PR TITLE
Bug 1192055 - Triggering shb events after removing screen class r=mhenretty

### PR DESCRIPTION
**Pull request by albertopq**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=dd6eb939579de4382bb2f6933dc989199434da5e

File: apps/system/js/software_button_manager.js [hunk 1]
```diff
@@ -202,19 +201,16 @@
...
+        var element = this.element;
+        // status[0] = current status
+        // status[1] = previous status
+        var status = ['enabled', 'disabled'];
+        status = this.enabled ? status : status.reverse();
+
+        element.classList.toggle('visible', !!(this.enabled));
+        this.screenElement.classList.add('software-button-' + status[0]);
+        this.screenElement.classList.remove('software-button-' + status[1]);
+        window.dispatchEvent(new CustomEvent('software-button-' + status[0]));
```
Comment (mikehenrty) [inline apps/system/js/software_button_manager.js]: This is clever, but it's a little bit hard to read at first. Can we have something like:
```
var previousState = this.enabled ? 'software-button-disabled' : 'software-button-enabled';
var currentState = this.enabled ? 'software-button-enabled' : 'software-button-disabled';
this.screenElement.classList.add(currentState);
this.screenElement.classList.remove(previousState);
window.dispatchEvent(new CustomEvent(currentState));
```
I like the simplification, but the code looks more complex than it is currently.

Comment (albertopq) [inline apps/system/js/software_button_manager.js]: Yeah, I had the same before but it looked too repetitive to me... but you are right, is more readable. Thanks!

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d41dfac5cde422c086e09f1e3db33dc8c0588866

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0782ed6c62b4bed5f7983602d9df3ab48378b9c4

**Pull request merged**